### PR TITLE
Game Bootstrap and Game States

### PR DESCRIPTION
**Pull request by Danielmontesgil**
Added Bootstrap, GameManager, GameStateManager, ViewManager, GameStateBase, ViewBase, ControllerBase, ModelBase.

With this classes the game starts and they are the base for our UI.

### CODE REVIEW

File: Assets/Scripts/Constants.cs
Comment (cris204) [inline Assets/Scripts/Constants.cs:1]: I don't know how good it is to move things from assetpaths to constant .

File: Assets/Scripts/GameStateManager.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using JueguitosPro.GameStates;
+
+namespace JueguitosPro
+{
+    public class GameStateManager
+    {
+        private Stack<IGameState> _gameStates = new();
```
Comment (cris204) [inline Assets/Scripts/GameStateManager.cs]: Please explain to me when are the variables with "_" at the beginning

File: Assets/Scripts/ViewManager.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            _mainUI = GameObject.FindObjectOfType<MainUI>();
+        }
+        
+        public T InstantiateView<T>(string prefabPath, MainUI.CanvasLayer canvasLayer, Action<T> callback = null) where T : ViewBase
+        {
+            RectTransform parent = _mainUI.GetCanvasLayer(canvasLayer);
+
+            var prefab = GetPrefabFromResources<T>(prefabPath);
+            
+            var view = GameObject.Instantiate(prefab);
```
Comment (cris204) [inline Assets/Scripts/ViewManager.cs:22]: The "var"

**Review by cris204 — APPROVED** [review]

File: Assets/Scripts/GameStateManager.cs (hunk 1 above)
Comment (Danielmontesgil) [reply Assets/Scripts/GameStateManager.cs]: They are private members but I will fix them

File: Assets/Scripts/ViewManager.cs (hunk 2 above)
Comment (Danielmontesgil) [reply Assets/Scripts/ViewManager.cs:22]: It is T, we said that for generic types is better to use var

**Pull request merged**